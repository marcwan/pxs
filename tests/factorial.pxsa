MODULE_BODY:
        DECLARE         idx
        SET             idx, 4
        DECLARE         result
        SET             result, 1
        DECLARE         i
        SET             i, 0
__FOR_LOOP_12345_TEST:
        DECLARE         ::__TMP1_::
        SUB             idx, 1, ::__TMP1_::
        DECLARE         ::__TMP2__::
        TESTGTE         i, ::__TMP1_::, ::__TMP2__::
        REMOVEVAR       ::__TMP1_::
        JUMPF           ::__TMP2__::, __FOR_LOOP_12345_END
        DECLARE         ::__TMP3__::
        SUB             idx, i, ::__TMP3__::
        MUL             result, ::__TMP3__::, result
        REMOVEVAR       ::__TMP3__::
__FOR_LOOP_12345_INCR:
        ADD             i, 1, i
        JUMP            __FOR_LOOP_12345_TEST
__FOR_LOOP_12345_END:
        DEBUGPRINT      result
        EXIT            1
END_MODULE_BODY
