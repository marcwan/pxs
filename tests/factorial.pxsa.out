DEBUGPRINT: result = 24.000000
